File: core/src/main/java/com/larryhsiao/nyx/core/tags/Tag.java
package com.larryhsiao.nyx.core.tags;

/**
 * Tag of Jot
 */
public interface Tag {
    /**
     * The title of this tag
     */
    String title();

    /**
     * The id of this Tag
     */
    long id();

    /**
     * The version of this tag
     */
    int version();
}

File: core/src/main/java/com/larryhsiao/nyx/core/tags/Tag.kt
package com.larryhsiao.nyx.core.tags

/**
 * Tag of Jot
 */
interface Tag {
    /**
     * The title of this tag
     */
    fun title(): String?

    /**
     * The id of this Tag
     */
    fun id(): Long

    /**
     * The version of this tag
     */
    fun version(): Int

    /**
     * This tag is marked as deleted
     */
    fun deleted(): Boolean
}